mod/state-transition/pkg/core/state_processor_staking.go

```diff
@@ -87,24 +107,7 @@ func (sp *StateProcessor[
 	st BeaconStateT,
 	dep DepositT,
 ) error {
-	var nextDepositIndex uint64
```

Comment:
for genesis this check has been moved to the client
for non-genesis block this check is redundant (caller already checks everything there is to check about deposits (but signature, done later))